tests/python/relay/test_ir_nodes.py

```diff
@@ -32,44 +44,58 @@ def test_tensor_type():
     assert tt.span == None
     str(tt)
 
+    # roundtrip preserves alpha-equality
+    back = json_roundtrip(tt)
+    assert back == tt
+
 
 def test_type_param():
     tp = relay.TypeParam('name', relay.Kind.Type)
     assert tp.kind == relay.Kind.Type
     # assert tp.span  # TODO allow us to set span
     str(tp)
 
+    back = json_roundtrip(tp)
+    # pointer equality will not be preserved so alpha-equality will fail
+    assert back.kind == tp.kind
+
 
 def test_func_type():
     type_params = tvm.convert([])
     type_constraints = tvm.convert([])  # TODO: fill me in
     arg_types = tvm.convert([])
-    ret_type = None
+    ret_type = relay.TensorType((1, 2, 3), 'float32')
     tf = relay.FuncType(arg_types, ret_type, type_params, type_constraints)
     assert tf.type_params == type_params
     assert tf.type_constraints == type_constraints
     assert tf.arg_types == arg_types
     assert tf.ret_type == ret_type
     assert tf.span == None
-    # TODO make sure we can set
+    # TODO make sure we can set span
     str(tf)
 
+    back = json_roundtrip(tf)
+    assert back == tf
+
 
 def test_tuple_type():
-    tp = relay.TypeParam('tp', relay.Kind.Type)
-    tf = relay.FuncType(tvm.convert([]), None, tvm.convert([]), tvm.convert([]))
+    # tp = relay.TypeParam('tp', relay.Kind.Type)
```

Comment:
Had to remove type parameters from here and other places because alpha-equality relied on pointer identity. Is there any way to handle that when serializing?